<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DefineShaderDialog</class>
 <widget class="QDialog" name="DefineShaderDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>690</width>
    <height>611</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Define Shader Dialog</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="0" column="0">
    <widget class="QLabel" name="label_3">
     <property name="text">
      <string>Predefined Shaders:</string>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QComboBox" name="cmbShaderList">
     <property name="layoutDirection">
      <enum>Qt::LeftToRight</enum>
     </property>
    </widget>
   </item>
   <item row="0" column="2">
    <spacer name="horizontalSpacer_2">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>490</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="1" column="0" colspan="3">
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tabVertexShader">
      <attribute name="title">
       <string>Vertex Shader</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Enter your code for the vertex shader here.</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <widget class="ShaderEditor" name="txtVertexShaderCode"/>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabFragmentShader">
      <attribute name="title">
       <string>Fragment Shader</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Enter your code for the fragment shader here.</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="ShaderEditor" name="txtFragmentShaderCode"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabShaderVariables">
      <attribute name="title">
       <string>Shader Variables</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QPushButton" name="cmdAddShaderVariable">
         <property name="text">
          <string>Add Shader Variable</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="cmdRemoveShaderVariable">
         <property name="text">
          <string>Remove Shader Variable</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>340</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="0" colspan="3">
        <widget class="QTreeWidget" name="lstShaderVariables">
         <property name="frameShape">
          <enum>QFrame::Box</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Plain</enum>
         </property>
         <property name="indentation">
          <number>0</number>
         </property>
         <property name="rootIsDecorated">
          <bool>false</bool>
         </property>
         <column>
          <property name="text">
           <string>Variable Name</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Type</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="2" column="2">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="buttonBox">
       <property name="text">
        <string>Ok</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ShaderEditor</class>
   <extends>QPlainTextEdit</extends>
   <header location="global">ShaderEditor.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>txtVertexShaderCode</tabstop>
  <tabstop>txtFragmentShaderCode</tabstop>
  <tabstop>cmdAddShaderVariable</tabstop>
  <tabstop>cmdRemoveShaderVariable</tabstop>
  <tabstop>lstShaderVariables</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>clicked()</signal>
   <receiver>DefineShaderDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>628</x>
     <y>574</y>
    </hint>
    <hint type="destinationlabel">
     <x>495</x>
     <y>605</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
